smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -484,43 +623,47 @@ contract BridgeBank is BankStorage,
         );
     }
 
+    /**
+     * @dev Burns a token
+     * @param recipient Bytes representation of destination address
+     * @param tokenAddress Token address
+     * @param tokenAmount How much should be burned
+     * @param _lockBurnNonce A global nonce for locking an burning tokens
+     */
     function _burnTokens(
         bytes calldata recipient,
         address tokenAddress,
         uint256 tokenAmount,
         uint256 _lockBurnNonce
     ) private onlyCosmosTokenWhiteList(tokenAddress) validSifAddress(recipient) {
         BridgeToken tokenToTransfer = BridgeToken(tokenAddress);
+        
         // burn tokens
-        tokenToTransfer.burnFrom(
-            msg.sender,
-            tokenAmount
-        );
+        tokenToTransfer.burnFrom(msg.sender, tokenAmount);
+
+        // revert if the token doesn't have a denom
+        string memory denom = getDenom(tokenAddress);
+        require(keccak256(abi.encodePacked(denom)) != keccak256(abi.encodePacked("")), "INV_DENOM");
```

Comment:
Add an if statement here that checks if the bridgetoken address is equal to the hardcoded erowan token address, if it is, then don't do that require statement and assign the denom to `rowan`, else go into a block that has the require statement in it.

Reply:
This will only be able to be tested off of a mainnet fork.